libraries/AP_HAL/Scheduler.cpp

```diff
@@ -22,3 +23,13 @@ void Scheduler::call_delay_cb()
     _delay_cb();
     _in_delay_callback = false;
 }
+
+ExpectDelay::ExpectDelay(const AP_HAL::HAL &hal, uint32_t ms) : _hal(hal)
```

Comment:
the way I originally structured it as a header-only macro it couldn't access the hal. I've restructured it now, and it can access the hal, so I'll remove the argument